vendor/k8s.io/kubernetes/pkg/kubectl/cmd/util/factory_object_mapping.go

```diff
@@ -77,21 +77,25 @@ func NewObjectMappingFactory(clientAccessFactory ClientAccessFactory) ObjectMapp
 // the built in mapper if necessary. It supports unstructured objects either way, since
 // the underlying Scheme supports Unstructured. The mapper will return converters that can
 // convert versioned types to unstructured and back.
+// It can return an error and the best effort unstructured mapper and typer.
 func (f *ring1Factory) objectLoader() (meta.RESTMapper, runtime.ObjectTyper, error) {
 	discoveryClient, err := f.clientAccessFactory.DiscoveryClient()
 	if err != nil {
-		glog.V(3).Infof("Unable to get a discovery client to find server resources, falling back to hardcoded types: %v", err)
-		return legacyscheme.Registry.RESTMapper(), legacyscheme.Scheme, nil
+		unstructuredMapper := discovery.NewRESTMapper(nil, meta.InterfacesForUnstructured)
+		unstructuredTyper := discovery.NewUnstructuredObjectTyper(nil)
+		return unstructuredMapper, unstructuredTyper, err
 	}
 
 	groupResources, err := discovery.GetAPIGroupResources(discoveryClient)
 	if err != nil && !discoveryClient.Fresh() {
 		discoveryClient.Invalidate()
 		groupResources, err = discovery.GetAPIGroupResources(discoveryClient)
 	}
+	// even if we can't get all the results, we're better off continuing with what we can than not presenting
+	// anything.  This mirrors old behavior that used to fallback to a hardcoded list of API types compiled
+	// into kubernetes.
 	if err != nil {
-		glog.V(3).Infof("Unable to retrieve API resources, falling back to hardcoded types: %v", err)
-		return legacyscheme.Registry.RESTMapper(), legacyscheme.Scheme, nil
```

Comment:
Error is happening here: when we run `oc process` expecting not to contact the server, returning a mapper for hardcoded types would suffice. With this patch, we are actually continuing on, and using a deferred discovery mapper down below.